### PR TITLE
Gdsii layers

### LINKED ISSUE #514: GDSII file introspection
Would be good to have some introspection functions for GDSII files, e.g. a function to return a list of layer numbers present in a file.

### PR DESCRIPTION
**Pull request by HomerReid**
Fixes #514. Requires most recent repository version of `libGDSII.`

````python
>>> mp.GDSII_layers('python/examples/coupler.gds')              
Out[2]: [0, 1, 2, 3, 4, 5, 31, 32]
```

### CODE REVIEW

File: python/meep.i [hunk 1]
```diff
@@ -840,33 +840,6 @@ meep::volume_list *make_volume_list(const meep::volume &v, int c,
...
-%typemap(in, fragment="NumPy_Macros") double* ztics {
-    $1 = (double *)array_data($input);
-}
-%typecheck(SWIG_TYPECHECK_POINTER, fragment="NumPy_Fragments") double* weights {
-    $1 = is_array($input);
-}
-%typemap(in, fragment="NumPy_Macros") double* weights {
-    $1 = (double *)array_data($input);
-}
-
```
Comment (stevengj) [inline python/meep.i]: Why are these lines deleted?

Comment (HomerReid) [reply python/meep.i]: I'm sorry.
